pkg/scheduler/factory/factory.go

```diff
@@ -465,13 +475,31 @@ func getBinderFunc(client clientset.Interface, extenders []algorithm.SchedulerEx
 	}
 }
 
-func (c *Configurator) getPriorityFunctionConfigs(priorityKeys sets.String) ([]priorities.PriorityConfig, error) {
+func (c *Configurator) getPriorityConfigs(priorityKeys sets.String) ([]priorities.PriorityConfig, *config.Plugins, []config.PluginConfig, error) {
+	var priorityConfigs []priorities.PriorityConfig
 	pluginArgs, err := c.getPluginArgs()
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
+	}
+
+	initialConfigs, err := getPriorityFunctionConfigs(priorityKeys, *pluginArgs)
+	if err != nil {
+		return nil, nil, nil, err
 	}
 
-	return getPriorityFunctionConfigs(priorityKeys, *pluginArgs)
+	var plugins config.Plugins
+	var pluginConfig []config.PluginConfig
+	for _, pc := range initialConfigs {
+		producer, ok := priorityToFrameworkConfigProducer[pc.Name]
```

Comment:
Should we transfer the weight? Unless we want to multiply it internally.

Reply:
Done.